test/test_maintenance_period.py

```diff
@@ -0,0 +1,124 @@
+"""
+
+import time
+from datetime import datetime, timedelta
+from alignak.misc.serialization import unserialize
+from alignak.downtime import Downtime
+from alignak.objects.timeperiod import Timeperiod
+
+from alignak_test import AlignakTest, unittest
+
+class TestMaintenancePeriod(AlignakTest):
+    """
+    This class tests the maintenance_period
+    """
+    def setUp(self):
+        """
+        For each test load and check the configuration
+        :return: None
+        """
+        self.print_header()
+        self.setup_with_file('cfg/cfg_default.cfg')
+        assert self.conf_is_correct
+
+        # Our scheduler
+        self._sched = self.schedulers['scheduler-master'].sched
+
+        # Our broker
+        self._broker = self._sched.brokers['broker-master']
+
+        # No error messages
+        assert len(self.configuration_errors) == 0
+        # No warning messages
+        assert len(self.configuration_warnings) == 0
+
+    def test_maintenance_period_host(self):
+        """Test a host enter in maintenance_period
+        
+        :return: None
+        """
+        self.print_header()
+        # Get the host
+        host = self._sched.hosts.find_by_name("test_host_0")
+        host.checks_in_progress = []
+        host.act_depend_of = []
+        # Not any downtime yet !
+        assert host.downtimes == {}
+
+        # Make the host be UP
+        self.scheduler_loop(1, [[host, 0, 'UP']])
+
+        # we created a new timeperiod from now -5 minutes to now + 55 minutes
+        begin = datetime.now() - timedelta(minutes=5)
+        end = datetime.now() + timedelta(minutes=55)
+
+        h_begin = format(begin, '%H:%M')
+        if format(begin, '%H') == '23' and format(begin, '%M') >= 55:
+            h_begin = '00:00'
+        h_end = format(end, '%H:%M')
+        end = end - timedelta(seconds=int(format(end, '%S')))
+        timestamp_end = int(time.mktime(end.timetuple()))
+
+        data = {
+            'timeperiod_name': 'maintenance',
+            'sunday': h_begin + '-' + h_end,
+            'monday': h_begin + '-' + h_end,
+            'tuesday': h_begin + '-' + h_end,
+            'wednesday': h_begin + '-' + h_end,
+            'thursday': h_begin + '-' + h_end,
+            'friday': h_begin + '-' + h_end,
+            'saturday': h_begin + '-' + h_end
+        }
+        timeperiod = Timeperiod(data)
+        timeperiod.explode()
+        self.schedulers['scheduler-master'].sched.timeperiods[timeperiod.uuid] = timeperiod
+        host.maintenance_period = timeperiod.uuid
+
+        # Make the host be UP again
+        self.scheduler_loop(1, [[host, 0, 'UP']])
+
+        assert 1 == len(host.downtimes)
+        # The service is still in a downtime period
+        assert host.in_scheduled_downtime
+        downtime = host.downtimes.values()[0]
+        assert downtime.fixed
+        assert downtime.is_in_effect
+        assert not downtime.can_be_deleted
+        assert downtime.end_time == timestamp_end
+
```

Comment:
Is there any specific downtime comment for the maintenance period? If so, we should test this comment value, no?